consensus/service/src/tx_manager/mod.rs

```diff
@@ -223,25 +158,18 @@ impl<E: ConsensusEnclave, UI: UntrustedInterfaces> TxManager<E, UI> {
         expired.keys().cloned().collect()
     }
 
-    /// Returns elements of `tx_hashes` that are not inside the cache.
-    pub fn missing_hashes<T>(&self, tx_hashes: &T) -> Vec<TxHash>
-    where
-        for<'a> &'a T: IntoIterator<Item = &'a TxHash>,
```

Comment:
mockall doesn't like this higher-ranked trait bound (it lets you pass a reference to any iterable collection of TxHashes as a function argument). I'm guessing that the simpler `contains` will work OK, but I will also check if the extra locking causes performance problems.

Reply:
Yup, would be interesting to see what effect this has. I think this is going to move us from one lock per SCP statement, so N locks per SCP statement, where N is the number of tx hashes the statement refers to.

Reply:
https://stackoverflow.com/questions/3652056/how-efficient-is-locking-an-unlocked-mutex-what-is-the-cost-of-a-mutex

should we place bets? :)